packages/contracts-bedrock/src/L2/SuperchainWETH.sol

```diff
@@ -9,21 +9,27 @@ import { Predeploys } from "src/libraries/Predeploys.sol";
 
 // Interfaces
 import { ISemver } from "src/universal/interfaces/ISemver.sol";
-import { IL2ToL2CrossDomainMessenger } from "src/L2/interfaces/IL2ToL2CrossDomainMessenger.sol";
 import { IL1Block } from "src/L2/interfaces/IL1Block.sol";
 import { IETHLiquidity } from "src/L2/interfaces/IETHLiquidity.sol";
-import { ISuperchainWETH } from "src/L2/interfaces/ISuperchainWETH.sol";
+import { ICrosschainERC20 } from "src/L2/interfaces/ICrosschainERC20.sol";
+import { Unauthorized, NotCustomGasToken } from "src/libraries/errors/CommonErrors.sol";
 
 /// @custom:proxied true
 /// @custom:predeploy 0x4200000000000000000000000000000000000024
 /// @title SuperchainWETH
 /// @notice SuperchainWETH is a version of WETH that can be freely transfrered between chains
 ///         within the superchain. SuperchainWETH can be converted into native ETH on chains that
 ///         do not use a custom gas token.
-contract SuperchainWETH is WETH98, ISuperchainWETH, ISemver {
+contract SuperchainWETH is WETH98, ICrosschainERC20, ISemver {
+    /// @notice A modifier that only allows the SuperchainTokenBridge to call
+    modifier onlySuperchainTokenBridge() {
+        if (msg.sender != Predeploys.SUPERCHAIN_TOKEN_BRIDGE) revert Unauthorized();
+        _;
+    }
```

Comment:
A while ago we came up with an internal best practice to only use modifiers if you have code both before and after the `_;`, such as in a reentrancy lock. From [this doc](https://www.notion.so/oplabs/Review-use-of-modifiers-24a8fc642aad4706a69bd4a9f9710736?pvs=4) which is probably private:

> Issues with modifiers:
> 
> 1. **Behavior Ambiguity:** When seeing a function signature like the first one, you can’t easily tell in what order code within each modifier is executed, and if the two might have logic that affects each other. With the second one it’s much more obvious. 
> ```solidity
> function foo() public modifier1 modifier2 {
>   // logic here -- we can't tell when modifiers are executed
>   //  unless we view their implementations
> }
> 
> function foo() public {
>   check1();
> 	// logic here
>   check2();
> }
>```
>         
> 2. **Readability:** In my opinion, modifiers are also easier to miss when reviewing because they are inlined
> 3. **Modifiers can’t modify inputs:** Not sure if we use them for this purpose (this is more of a common use case in tests) but modifiers can’t return values and therefore they can’t be used to bound or adjust inputs. A common use case for this is adjusting sentinel values, e.g. for convenience you may let the user pass `amount == type(uint256).max` which gets adjusted to `amount = token.balanceOf(msg.sender)`
> 
> The only time to use modifiers is when the modifier must do something both before AND after function execution, such as a reentrancy guard. In other cases, an inline internal function is preferred. See [this twitter thread](https://twitter.com/DrakeEvansV1/status/1708271596207579341) for more info